src/main/java/com/google/sps/app/servlets/LogIn.java

```diff
@@ -0,0 +1,62 @@
+package com.google.sps.app.servlets;
+
+import java.io.IOException;
+import java.util.UUID;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.cloud.datastore.BaseEntity;
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.Key;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.sps.util.Validator;
+
+@WebServlet("/LogIn")
+public class LogIn extends HttpServlet {
+
+    final String textInput = "text-input-";
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String username = request.getParameter(textInput+"user-name").trim();
+        if(!Validator.validUserName(username)) {
+            response.getWriter().println("Invalid Username");
+            return;
+        }
+
+        long userId;
+        try {
+            userId = queryDatastore(username);
+        } catch (IllegalArgumentException e) {
+            response.getWriter().println(e.getMessage());
+            return;
+        }
+
+        response.getWriter().println(userId);
+    }
+
+    public long queryDatastore(String username) throws IllegalArgumentException {
+
+        Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+
+        String gqlQuery = "select * from User where username=" + username;
```

Comment:
Make this a const too. Also if you only need the id, you can get just the Id.

private static final USER_QUERY_TEMPLATE =
"SELECT userId FROM User WHERE username=";

Reply:
Oh nice. I was doing some googling but couldn't figure it quite out. Thank you.

Reply:
How do I treat the result. Is it an int?

Reply:
I tried the Query  with your Suggestion database and I got an error. Maybe some additional index is required
![image](https://user-images.githubusercontent.com/50251777/177017696-995a3c37-bf24-4f14-b329-a208c48a7535.png)

Reply:
Maybe I am wrong. Looks like datastore might not let you fetch specific column. Leave it as it is for now. I will see if it is possible

Reply:
You need to set up an index. You can investigate and do that later.